### DISCUSSION TITLE
Video mp4 transcribe behave differently in playground vs API

### DISCUSSION THREAD

**Discussion by stefanosalvucci** [deepgram/community#178] [General help] [answered]
### Which Deepgram product are you using?

Deepgram API

### Details

Hello, if I transcribe an mp4 video on the playground (https://playground.deepgram.com/) the trascription looks incredibly fine.
However, if I try to do the same request through API,:

POST https://api.deepgram.com/v1/listen?punctuate=true&paragraphs=true&language=it&model=enhanced
{ 'url' => file_url }
{ 'Authorization' => 'Token ' + ENV['DEEPGRAM_API_KEY'], 'Content-Type' => 'application/json'}

==================================

the response looks totally different from the one shown in playground
Here's the response through playground VS API
**PGROUND**

```
{
  "metadata": {
    "transaction_key": "deprecated",
    "request_id": "9a343bb9-5e2b-4c54-adf8-c4c0209338ab",
    "sha256": "f82c04ee8292c84354ae3c378f564298e3e3f273776e35435fb7db84810dce23",
    "created": "2023-06-01T15:26:53.607Z",
    "duration": 31.850687,
    "channels": 1,
    "models": [
      "62e651ac-471f-4b22-8dfc-70a21049b274"
    ],
    "model_info": {
      "62e651ac-471f-4b22-8dfc-70a21049b274": {
        "name": "general-enhanced",
        "version": "1982-10-20.26461",
        "arch": "polaris"
      }
    }
  },
  "results": {
    "channels": [
      {
        "alternatives": [
          {
            "transcript": "Ciao sono stefano salvucci sono il tuo docente di questo modulo del master Oggi parleremo di soft skill. Che cosa sono? Competenze non tecniche fondamentali per riuscire ad avere successo ed andare avanti in questo mondo dell'informatica. Ti auguro una buona lezione e ti ricordo che all'interno troverai dei contenuti che ho selezionato per te e che ritengo molto molto validi buona lezione",
            "confidence": 0.99316406,
            "words": [
              {
                "word": "ciao",
                "start": 4.31784,
                "end": 4.7176404,
                "confidence": 0.80810547,
                "punctuated_word": "Ciao"
              },
    .........
              {
                "word": "lezione",
                "start": 30.05282,
                "end": 30.55282,
                "confidence": 0.9980469,
                "punctuated_word": "lezione"
              }
            ],
            "paragraphs": {
              "transcript": "\nCiao sono stefano salbucci sono il tuo docente di questo modulo del master Oggi parleremo di soft skill. Che cosa sono? Competenze non tecniche fondamentali per riuscire ad avere successo ed andare avanti in questo mondo dell'informatica. Ti auguro una buona lezione e ti ricordo che all'interno troverai dei contenuti che ho selezionato per te e che ritengo molto molto validi buona lezione",
              "paragraphs": [
                {
                  "sentences": [
                    {
                      "text": "Ciao sono stefano salbucci sono il tuo docente di questo modulo del master Oggi parleremo di soft skill.",
                      "start": 4.31784,
                      "end": 10.60428
                    },
                    {
                      "text": "Che cosa sono?",
                      "start": 10.7039795,
                      "end": 11.723721
                    },
                    {
                      "text": "Competenze non tecniche fondamentali per riuscire ad avere successo ed andare avanti in questo mondo dell'informatica.",
                      "start": 12.14326,
                      "end": 19.56838
                    },
                    {
                      "text": "Ti auguro una buona lezione e ti ricordo che all'interno troverai dei contenuti che ho selezionato per te e che ritengo molto molto validi buona lezione",
                      "start": 20.54764,
                      "end": 30.55282
                    }
                  ],
                  "num_words": 63,
                  "start": 4.31784,
                  "end": 30.55282
                }
              ]
            }
          }
        ]
      }
    ]
  }
}
```

**API**
`{"metadata":{"transaction_key":"deprecated","request_id":"76c9ab87-d5f4-4de8-9600-c50f661d6a8f","sha256":"f82c04ee8292c84354ae3c378f564298e3e3f273776e35435fb7db84810dce23","created":"2023-06-01T15:25:34.093Z","duration":31.850687,"channels":1,"models":["96a295ec-6336-43d5-b1cb-1e48b5e6d9a4"],"model_info":{"96a295ec-6336-43d5-b1cb-1e48b5e6d9a4":{"name":"general","version":"2023-02-22.3","arch":"base"}}},"results":{"channels":[{"alternatives":[{"transcript":"model master","confidence":0.81347656,"words":[{"word":"model","start":7.419954,"end":7.919954,"confidence":0.81347656},{"word":"master","start":7.9754586,"end":8.475458,"confidence":0.6015625}]}]}]}}`

=================

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

https://api.deepgram.com/v1/listen?punctuate=true&paragraphs=true&language=it&model=enhanced

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

9a343bb9-5e2b-4c54-adf8-c4c0209338ab

**Comment by jjmaldonis** [comment]
Hey @stefanosalvucci, thanks for the detailed information.

We often see results like this if the `language` parameter is set to a different language than the audio. Is it possible that the API call, for some reason, did not include the `language=it` parameter? Given that the API call's result has a transcript of `"model master"`, which are both English words, my gut tells me that transcript was being generated using English rather than Italian. Also, the Deepgram API defaults to using English, so if the `language` parameter isn't in the API request then English will be used.

Alternatively: I tried to reproduce the issue by running the same API call (https://api.deepgram.com/v1/listen?punctuate=true&paragraphs=true&language=it&model=enhanced) on an Italian podcast. The transcript I received looks good to me, so it's possible that we'll only be able to reproduce the issue with the audio file you're using. Could you please upload the audio file (or send us a link to it) so we can see if it's an issue with that specific audio file?

**Comment by stefanosalvucci** [comment]
Hi @jjmaldonis, here you can find the video: https://file.io/orWmX522W2EE
It's actually an .mp4 video, but since it was working on the playground I tough that should work also on API call

**Comment by stefanosalvucci** [comment, marked as answer]
After some debugging and thanks to https://console.deepgram.com/project/XXXXXX/usage/logs I've discovered that the wrapper for POST request was inserting the query string as body in the request.

Sorry for the disturb
